jax/_src/numpy/ufunc_api.py

```diff
@@ -32,6 +33,37 @@
 import numpy as np
 
 
+def get_if_single_primitive(fun, *args):
+  """
+  If fun(*args) lowers to a single primitive with inputs and outputs matching function
+  inputs and outputs, return that primitive. Otherwise return None.
+  """
+  try:
+    jaxpr = jax.make_jaxpr(fun)(*args)
+  except:
+    return None
+  while len(jaxpr.eqns) == 1:
+    if jaxpr.eqns[0].primitive == jax._src.pjit.pjit_p:
+      jaxpr = jaxpr.eqns[0].params['jaxpr']
```

Comment:
I think this logic isn't quite right because it'll skip checking whether the invars/outvars are arranged correctly in the outer jaxpr. We should probably only recurse if we check those invars/outvars also.

Reply:
Nice catch again!

I modified the logic, and added a new `non_fastpath` test case that catches this issue.